docs/docs/config/kscan.md

```diff
@@ -0,0 +1,385 @@
+        <tr><th>Row 0</th><td>A0</td><td>A1</td><td>A2</td></tr>
+        <tr><th>Row 1</th><td>A3</td><td>A4</td><td>A5</td></tr>
+        <tr><th>Row 2</th><td>A6</td><td>A7</td><td>A8</td></tr>
+        <tr><th>Row 3</th><td>B0</td><td>B1</td><td>(none)</td></tr>
+    </tbody>
+</table>
+
+...which can be configured with the following Devicetree code:
+
+```devicetree
+/ {
+    chosen {
+        zmk,kscan = &kscan0;
+    };
+
+    kscan0: kscan_composite {
+        compatible = "zmk,kscan-composite";
+        label = "KSCAN0";
+        rows = <3>;
+        columns = <4>;
+
+        // Include the matrix driver
+        matrix {
+            kscan = <&kscan1>;
+        };
+
+        // Include the direct GPIO driver...
+        direct {
+            kscan = <&kscan2>;
+            row-offset = <3>; // ...and shift it to not overlap
+        };
+    };
+
+    kscan1: kscan_matrix {
+        compatible = "zmk,kscan-gpio-matrix";
+        // define 3x3 matrix here...
+    };
+
+    kscan2: kscan_direct {
+        compatible = "zmk,kscan-gpio-direct";
+        // define 2 direct GPIOs here...
+    };
+}
+```
+
+## Mock Driver
+
+Mock keyboard scan driver that simulates key events.
+
+### Devicetree
+
+Applies to: `compatible = "zmk,kscan-mock"`
+
+Definition file: [zmk/app/dts/bindings/zmk,kscan-mock.yaml](https://github.com/zmkfirmware/zmk/blob/main/app/dts/bindings/zmk%2Ckscan-mock.yaml)
+
+| Property       | Type   | Description                                   | Default |
+| -------------- | ------ | --------------------------------------------- | ------- |
+| `label`        | string | Unique label for the node                     |         |
+| `event-period` | int    | Milliseconds between each generated event     |         |
+| `events`       | array  | List of key events to simulate                |         |
+| `rows`         | int    | The number rows of in the composite matrix    |         |
+| `cols`         | int    | The number columns of in the composite matrix |         |
+| `exit-after`   | bool   | Exit the program after running all events     | false   |
+
+The `events` array should be defined using the macros from [dt-bindings/zmk/kscan_mock.h](https://github.com/zmkfirmware/zmk/blob/main/app/include/dt-bindings/zmk/kscan_mock.h).
+
+## Matrix Transform
+
+Defines a mapping from keymap logical positions to physical matrix positions.
+
+Transforms should be used any time the physical layout of a keyboard's keys does not match the layout of its electrical matrix and/or when not all positions in the matrix are used. This applies to most non-ortholinear boards.
+
+Transforms can also be used for keyboards with multiple layouts. You can define multiple matrix transform nodes, one for each layout, and users can select which one they want from the `/chosen` node in their keymaps.
+
+See the [new shield guide](../development/new-shield.md#optional-matrix-transform) for more documentation on how to define a matrix transform.
```

Comment:
I think this section is pretty great and it would be really helpful to have a link from the linked new shield section back to here.